src/framework/core/js/ResourceLoader.js

```diff
@@ -733,6 +733,10 @@ https://github.com/fluid-project/infusion/raw/main/Infusion-LICENSE.txt
 
     fluid.resourceLoader.loaders.resourceText.noPath = true;
 
+    // Implementation note: Whilst there will not be a type error if the user writes a non-promise return, there will
+    // be a race failure as the "resource" initialiser in fluid.resourceFromRecord loses to the transformEvent listener,
+    // overwriting its result with the FetchOne value as, e.g. fluid.getForComponent resolves synchronously. As a result
+    // we have to say we can't support synchronous "early" resources.
```

Comment:
Isn't it possible to wrap this in a promise, or is it the fact that it would still resolve immediately a problem?

Reply:
Yes, the race is to do with timing rather than the return type, but thanks for drawing my attention back to this comment which I can now delete/update since the issue is resolved